R/mlp-fit.R

```diff
@@ -634,6 +658,9 @@ mlp_fit_imp <-
     # Optimize parameters
     for (epoch in 1:epochs) {
 
+     learn_rate <- set_learn_rate(epoch - 1, learn_rate, type = rate_schedule, ...)
+     optimizer <- torch::optim_sgd(model$parameters, lr = learn_rate, momentum = momentum)
```

Comment:
Creating a new optimizer every epoch will reset the optimizer state. For example, SGD with momentum keeps track of the latest update for each parameter, so it might be OK to just restart. 

But other optimizers like Adam, will keep a moving average of updates and reseting them can interfere in training.

Reply:
I'd modify the learning rate with something like: 

```
for (i in seq_along(optimizer$param_groups)) {
   optimizer$param_groups[[i]]$lr <- new_lr
}
```

To avoid having to create a new optimizer every epoch. 
We we'll probably only have a single parameter group ever, so we could simplify this to:

```
optimizer$param_groups[[1]]$lr <- new_lr
```

Reply:
Ok, I'll leave as-is for now. SGD seems to work well with this version and I don't know that we'll add another optimizer, esp for these (relatively) simple networks.